### PR TITLE
Add YAML line selection highlighting in GShootEditor

### LINKED ISSUE #1961: Shareable shoot yaml snippets
**What would you like to be added**:

It will be really handy to be able to share shoot yaml snippets.

When clicking on the shoot yaml line in the editor the URL to change the same way how GitHub is doing it.
 
Something like that:
```
.../namespace/garden/shoots/<shoot-name>/yaml#L300
```
**Why is this needed**:

For convenience

### PR DESCRIPTION
**Pull request by petersutter**
**What this PR does / why we need it**:
This PR implements a feature to highlight selected lines in YAML files within the GShootEditor component.
The line selection is reflected in the URL.
Screenshot 2024-07-16 at 12 18 33

**Which issue(s) this PR fixes**:
Fixes #1961

**Special notes for your reviewer**:

**Release note**:

```feature user
Adds line selection highlighting in the `Shoot` YAML editor when selecting the line numbers. The line selection is reflected in the URL.
```

### CODE REVIEW

**Comment by grolu** [comment]
While this looks nice I'm asking myself - does this make sense? I mean the resource is constantly changing and the link is not pinned to a specific resource version. This means that the highlighted lines may be the wrong ones.
Most of the noise happens in status which is at the end of the yaml - but still this is a limitation users may not be aware of.

**Comment by petersutter** [comment]
inviting @StenlyTU into the [discussion](https://github.com/gardener/dashboard/pull/1982#issuecomment-2230667106)

**Comment by StenlyTU** [comment]
That was fast, thanks. I think it's worth it as long as we mimic the Github behaviour.

Let's have more opinions:
/invite @gardener/gardener-landscape-stability WDYT?

**Comment by dguendisch** [comment]
> does this make sense? I mean the resource is constantly changing and the link is not pinned to a specific resource version. This means that the highlighted lines may be the wrong ones.

I'd expect that only the `.status` field is changing constantly, while everything above should remain valid, so as long as you don't share some status links it should be fine.
I like it ❤️.

File: frontend/src/components/GShootEditor.vue [hunk 1]
```diff
@@ -308,6 +308,52 @@ onBeforeUnmount(() => {
 <style lang="scss">
   @import 'vuetify/settings';
 
+  .v-theme--light .g-highlighted-top {
```
[the remark this answers is not in the document]
Comment (petersutter) [reply frontend/src/components/GShootEditor.vue]: 👍 ✅

File: frontend/src/composables/useEditorLineHighlighter.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import {
+  computed,
+  ref,
+} from 'vue'
+import { useUrlSearchParams } from '@vueuse/core'
+
+export function useEditorLineHighlighter (cm) {
```
[the remark this answers is not in the document]
Comment (petersutter) [reply frontend/src/composables/useEditorLineHighlighter.js:13]: 👍 thanks ✅

File: frontend/src/composables/useEditorLineHighlighter.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      const values = line.split('-')
+        .map(value => parseInt(value, 10) - 1)
+        .sort((a, b) => a - b)
+      return values.every(Number.isInteger)
+        ? values
+        : []
+    },
+    set (values = []) {
+      const [startLine, endLine = startLine] = values
+      if (!values.length || !values.every(Number.isInteger)) {
```
[the remark this answers is not in the document]
Comment (petersutter) [reply frontend/src/composables/useEditorLineHighlighter.js]: I have no strong opinion in this case, I have pushed this change ✅

**Review by holgerkoser — APPROVED** [review]
/lgtm

**Review by grolu — APPROVED** [review]
/lgtm

**Pull request merged**